packages/cloudflare/src/cli/templates/cache-handler/cache-store.ts

```diff
@@ -0,0 +1,65 @@
+import type { IncrementalCacheValue } from "next/dist/server/response-cache";
+
+export type CacheEntry = {
+  lastModified: number;
+  value: IncrementalCacheValue | null;
+};
+
+export type CacheStore = {
+  get: (key: string) => Promise<CacheEntry | null>;
+  put: (key: string, entry: CacheEntry, ttl?: number) => Promise<void>;
+};
+
+export function getCacheStore() {
+  const kvName = process.env.__OPENNEXT_KV_BINDING_NAME;
+  if (process.env[kvName]) {
+    return new KVStore(process.env[kvName] as unknown as KVNamespace);
+  }
+
+  return new CacheAPIStore();
+}
+
+const defaultTTL = 31536000; // 1 year
+
+class KVStore implements CacheStore {
+  constructor(private store: KVNamespace) {}
+
+  get(key: string) {
+    return this.store.get<CacheEntry>(key, "json");
+  }
+
+  put(key: string, entry: CacheEntry, ttl = defaultTTL) {
+    return this.store.put(key, JSON.stringify(entry), {
+      expirationTtl: ttl,
+    });
+  }
+}
+
+class CacheAPIStore implements CacheStore {
```

Comment:
@petebacondarwin @dario-piotrowicz I'm not familiar the cf cache API, do you have thoughts about this class?

With my limited understanding of the cache API, I'm not sure if we should use it for the fetch cache (as long as the cache headers are set correctly on the `fetch` call).

I'm also wondering about [custom keys](Currently, it is not possible to purge a URL stored through Cache API that uses a custom cache key set by a Worker).